tendrl/gluster_integration/sds_sync/__init__.py

```diff
@@ -862,3 +820,59 @@ def brick_status_alert(hostname):
     finally:
         if isinstance(lock, etcd.lock.Lock) and lock.is_acquired:
             lock.release()
+
+
+def update_cluster_alert_count():
+    cluster_alert_count = 0
+    severity = ["WARNING", "CRITICAL"]
+    try:
+        alert_counts = find_volume_id()
+        alerts_arr = NS.tendrl.objects.ClusterAlert(
+            tags={'integration_id': NS.tendrl_context.integration_id}
+        ).load_all()
+        for alert in alerts_arr:
+            alert.tags = json.loads(alert.tags)
+            if alert.severity in severity:
+                cluster_alert_count += 1
+                if alert.resource in NS.gluster.objects.VolumeAlertCounters(
+                        )._defs['relationship'][alert.alert_type.lower()]:
+                    vol_name = alert.tags.get('volume_name', None)
+                    if vol_name:
+                        if vol_name in alert_counts.keys():
+                            alert_counts[vol_name]['alert_count'] += 1
+        # Update cluster alert count
+        NS.tendrl.objects.ClusterAlertCounters(
+            integration_id=NS.tendrl_context.integration_id,
+            alert_count=cluster_alert_count
+        ).save()
+        # Update volume alert count
+        for volume in alert_counts:
```

Comment:
I understand alert_counts is a dict. If so you should use something like

```
for vol_name, alert_det in alert_counts.iter_items():
    NS.gluster.objects.VolumeAlertCounters(
        integration_id=NS.tendrl_context.integration_id,
        alert_count=alert_det['alert_count'],
        volume_id=alert_det['vol_id']
    ).save()
```

Does this work. Even I feel the dict created in `find_volume_id` functions could be with keys as volume ids and value as counters. Why you need volume name as keys? Do you really use that anywhere?

Reply:
the alert object contains volume name only present in tags, to compare this alert with a particular volume i need volume name. that why i am creating dict with the volume name. During alert iteration, i am just comparing volume name from alert object with newly formed dict to increase the alert count.

Reply:
ok